labonneboite/common/search.py

```diff
@@ -235,21 +236,27 @@ def get_companies(self, add_suggestions=False):
         self.company_count = self._get_company_count(self.rome, self.distance)
         logger.debug("set company_count to %s from fetch_companies", self.company_count)
 
-        if self.from_number < 1:
-            self.from_number = 1
-            self.to_number = 10
-        if (self.from_number - 1) % 10:
-            self.from_number = 1
-            self.to_number = 10
+        max_page_size = pagination.OFFICES_MAXIMUM_PAGE_SIZE
+        current_page_size = self.to_number - self.from_number + 1
+        
+        invalid_pagination = (
+            self.from_number < 1
+            or current_page_size < 1  # this happens when a page out of bound is requested
+            or current_page_size > max_page_size
+        )
+
+        if invalid_pagination:
+            msg = "Invalid pagination error: max_page_size=%s from=%s to=%s company_count=%s" % (
+                max_page_size,
+                self.from_number,
+                self.to_number,
+                self.company_count,
+            )
+            raise pagination.InvalidPaginationArgument(msg)
```

Comment:
Ça m'embête cette exception levée et qui n'est pas intercepté nul part...

Reply:
en fait ce bloc ne sert à rien côté API car l'API construit correctement from et to à partir de page et page_size et fait déjà les contrôles

je vais donc déplacer ce bloc côté frontend WIP FIXME

Reply:
simplification done => see https://github.com/StartupsPoleEmploi/labonneboite/commit/df80b5de7bab1541c590ea5b2491643d52ac7e56